nifti2/nifti_tool.c

```diff
@@ -2323,20 +2323,25 @@ int act_add_exts( nt_opts * opts )
 /*----------------------------------------------------------------------
  * Return the allocated file contents.
  *----------------------------------------------------------------------*/
-static char * read_file_text(const char * filename, int64_t * length)
+static char * read_file_text(const char * filename, int * length)
 {
-   FILE * fp;
-   char * text;
+   FILE    * fp;
+   char    * text;
+   int64_t   len64;
+   size_t    bytes;
 
    if( !filename || !length ) {
       fprintf(stderr,"** bad params to read_file_text\n");
       return NULL;
    }
 
-   int64_t len = nifti_get_filesize(filename);
-   if( len <= 0 ) {
+   len64 = nifti_get_filesize(filename);
+   if( len64 <= 0 ) {
       fprintf(stderr,"** RFT: file '%s' appears empty\n", filename);
       return NULL;
+   } else if( len64 > INT_MAX ) {
```

Comment:
INT32_MAX would be more correct.

Reply:
Sure.  I almost never use those terms, but would expect INT_MAX to be reasonably appropriate when comparing with ints.  Either is fine.  We require int to mean int32_t.